### PR TITLE
Raw transaction for Nebulas

### PR DESCRIPTION
**Pull request by Woo-ing**
## Description
Adds new function for serialize the signed transaction.

## Testing instructions

We have added TransactionTests module to test new function.

## Types of changes

New feature (non-breaking change which adds functionality)

## Checklist

- [ ] Prefix PR title with `[WIP]` if necessary.
- [x] Add tests to cover changes as needed.
- [ ] Update documentation as needed.

### CODE REVIEW

File: js/tests/blockchain/nebulas/NebulasSigner.test.ts [hunk 1]
```diff
@@ -22,6 +21,7 @@ describe('NebulasSigner', () => {
 
         expect(output.algorithm == 1);
         expect(bufToHex(output.signature)).to.equal('0xf53f4a9141ff8e462b094138eccd8c3a5d7865f9e9ab509626c78460a9e0b0fc35f7ed5ba1795ceb81a5e46b7580a6f7fb431d44fdba92515399cf6a8e47e71500');
+        expect(output.raw == 'CiBQXdR2neMqnEu21q/U+OHqZHSBX9Q0hNiRfL2eCZO4hRIaGVefwtw23wEobqA40/7aIwQHghETxH4r+50aGhlXf89CeLWgHFjKu9/6tn4KNbelsMDAIIi2IhAAAAAAAAAAAJin2bgxTAAAKAcwyony5wU6CAoGYmluYXJ5QAFKEAAAAAAAAAAAAAAAAAAPQkBSEAAAAAAAAAAAAAAAAAADDUBYAWJB9T9KkUH/jkYrCUE47M2MOl14Zfnpq1CWJseEYKngsPw19+1boXlc64Gl5Gt1gKb3+0MdRP26klFTmc9qjkfnFQA=');
     });
 
-});
+});
```
Comment (hewigovens) [inline js/tests/blockchain/nebulas/NebulasSigner.test.ts]: We prefer to add a new line to the end of file, can you keep them across all changed files?

File: src/Coin.cpp [hunk 2]
```diff
@@ -300,7 +300,7 @@ std::string TW::deriveAddress(TWCoinType coin, const PublicKey& publicKey) {
 
     case TWCoinTypeWaves:
         return Waves::Address(publicKey).string();
-
+
```
Comment (hewigovens) [inline src/Coin.cpp]: remove extra spaces

File: src/Tron/Protobuf/TronInternal.pb.h [hunk 3]
```diff
@@ -2213,4 +2213,4 @@ inline const EnumDescriptor* GetEnumDescriptor< ::protocol::Transaction_Contract
 // @@protoc_insertion_point(global_scope)
 
 #include <google/protobuf/port_undef.inc>
-#endif  // PROTOBUF_INCLUDED_TronInternal_2eproto
+#endif  // PROTOBUF_INCLUDED_TronInternal_2eproto
```
Comment (hewigovens) [inline src/Tron/Protobuf/TronInternal.pb.h]: Can you revert unrelated file changes? this one and `ZilliqaMessage.pb.h`

File: src/proto/Nebulas.proto [hunk 4]
```diff
@@ -30,15 +30,36 @@ message SigningInput {
     // Timestamp to create transaction (256-bit number)
     bytes timestamp = 8;
 
-    // Optional payload
-    bytes payload = 9;
```
Comment (hewigovens) [inline src/proto/Nebulas.proto]: how to pass payload data from Swift/Kotlin now?

File: tests/interface/TWCoinTypeConfigTests.cpp [hunk 5]
```diff
@@ -467,7 +467,7 @@ TEST(TWCoinTypeConfiguration, TWCoinTypeConfigurationGetID) {
 
     auto nas = WRAPS(TWCoinTypeConfigurationGetID(TWCoinTypeNebulas));
     assertStringsEqual(nas, "nebulas");
-
+
```
Comment (hewigovens) [inline tests/interface/TWCoinTypeConfigTests.cpp]: remove extra spaces

File: tests/interface/TWNebulasTests.cpp [hunk 6]
```diff
@@ -54,4 +54,4 @@ TEST(Nebulas, AddressIsValidString) {
     EXPECT_FALSE(TWNebulasAddressIsValidString(STRING("a1V5bB2tbaM3FUiL4eRwpBLgEredS5C2wLY").get()));
     EXPECT_FALSE(TWNebulasAddressIsValidString(STRING("n2V5bB2tbaM3FUiL4eRwpBLgEredS5C2wLY").get()));
     EXPECT_FALSE(TWNebulasAddressIsValidString(STRING("n123").get()));
-}
```
Comment (hewigovens) [inline tests/interface/TWNebulasTests.cpp]: Keep new line

File: tests/Nebulas/TransactionTests.cpp [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+    auto to = Address("n1SAeQRVn33bamxN4ehWUT7JGdxipwn8b17");
+    auto transaction = Transaction(
+        /* to: */ from,
+        /* nonce: */ 7,
+        /* gasPrice: */ 1000000,
+        /* gasLimit: */ 200000,
+        /* to: */ to,
+        /* amount: */ 11000000000000000000ULL,
+        /* timestamp: */ 1560052938,
+        /* payload: */ std::string("{\"binary\":\"test\"}"));
```
Comment (hewigovens) [inline tests/Nebulas/TransactionTests.cpp]: payload is json string?

**Comment by hewigovens** [comment]
@Xu-Jianxin hi, can you fix codecov?

File: tests/Nebulas/TransactionTests.cpp (hunk 7 above)
Comment (Woo-ing) [reply tests/Nebulas/TransactionTests.cpp]: yes, payload is json string now. so we can support more extensions in the future.

**Comment by Woo-ing** [comment]
> @Xu-Jianxin hi, can you fix codecov?

ok
